core/src/main/scala/kafka/Kafka.scala

```diff
@@ -34,11 +34,20 @@ object Kafka extends Logging {
     val overrideOpt = optionParser.accepts("override", "Optional property that should override values set in server.properties file")
       .withRequiredArg()
       .ofType(classOf[String])
+    // This is just to make the parameter show up in the help output, we are not actually using this due the
+    // fact that this class ignores the first parameter which is interpreted as positional and mandatory
+    // but would not be mandatory if --version is specified
+    // This is a bit of an ugly crutch till we get a chance to rework the entire command line parsing
+    val versionOpt = optionParser.accepts("version", "Print version information.")
```

Comment:
nit: if we don't need the variable, maybe we can just remove it?

For some reason, this wasn't working for me. I looked in the `kafka-server-start.sh` script and found the following:

```
if [ $# -lt 1 ];
then
        echo "USAGE: $0 [-daemon] server.properties [--override property=value]*"
        exit 1
fi
```

Because of this check, we cannot get the proper help message if we don't specify any arguments. It seems we can just get rid of it.

Reply:
Hi Jason, I noticed that as well and opened [KAFKA-8247](https://issues.apache.org/jira/browse/KAFKA-8247) for it, as this seemed to me to broaden the scope of this PR too much beyond the related ticket.
I don't have any real preference though, happy to address it here as well.

Reply:
If we remove the variable, the --version flag won't show up in the _usage_ output, once we fix the duplicate (triplicate actually) parameter check. So it is not strictly speaking unused, just unused by our code, the argument parsing library looks at it.
If we decide to keep it we should also add the help parameter though. I've not changed anything yet, will either remove this or add help after we've discussed this. 
I'd prefer adding help, but that is not a strong preference.

Reply:
Thanks, I hadn't noticed KAFKA-8247, but I think it makes sense to address this there.